pkg/tfbridge/tests/provider_test.go

```diff
@@ -12,14 +12,171 @@ import (
 	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 	"github.com/stretchr/testify/assert"
 
+	sdkdiag "github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/pulumi/providertest/replay"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen"
 	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/stretchr/testify/require"
 )
 
+// Demonstrating the use of the newTestProvider helper.
+func TestTagsMap(t *testing.T) {
+
+	type testCase struct {
+		name               string
+		planResourceChange bool
+		doSetTags          bool
+		createTagsAs       map[string]string
+		createdProperties  any
+	}
+
+	testCases := []testCase{
+		{"basic", false, false, nil, map[string]any{"id": "*"}},
+		{"setnil", false, true, nil, map[string]any{"id": "*", "tags": map[string]any{}}},
+		{"setemp", false, true, map[string]string{}, map[string]any{"id": "*", "tags": map[string]any{}}},
+		{"setone", false, true, map[string]string{"a": "b"}, map[string]any{"id": "*", "tags": map[string]any{"a": "b"}}},
+
+		{"basic-prc", true, false, nil, map[string]any{"id": "*", "tags": nil}},                 // suspect
+		{"setnil-prc", true, true, nil, map[string]any{"id": "*", "tags": nil}},                 // suspect
+		{"setemp-prc", true, true, map[string]string{}, map[string]any{"id": "*", "tags": nil}}, // suspect
+		{"setone-prc", true, true, map[string]string{"a": "b"}, map[string]any{"id": "*", "tags": map[string]any{"a": "b"}}},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			p := newTestProvider(ctx, tfbridge.ProviderInfo{
+				P: shimv2.NewProvider(&schema.Provider{
+					Schema: map[string]*schema.Schema{},
+					ResourcesMap: map[string]*schema.Resource{
+						"example_resource": {
+							CreateContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) sdkdiag.Diagnostics {
+								rd.SetId("id0")
+								if tc.doSetTags {
+									rd.Set("tags", tc.createTagsAs)
```

Comment:
I'm a bit confused about this - shouldn't Create only be allowed to set Computed properties?

Reply:
That's an excellent point. I think that's exactly right documentation wise, but apparently this property is not enforced by the framework. The bridge is doing something really special because we changed how tags property work to remove Computed=true from the original upstream. I can confirm however that tagsInterceptor does in fact call rd.Set() in the bridge. And that seems to take effect and work.